crates/resolver/src/config.rs

```diff
@@ -480,21 +487,49 @@ pub struct ResolverOpts {
     ///
     /// If this is set, any positive responses with a TTL lower than this value will have a TTL of
     /// `positive_min_ttl` instead. Otherwise, this will default to 0 seconds.
+    #[cfg_attr(
+        feature = "serde",
+        serde(
+            serialize_with = "duration_opt_serialize",
+            deserialize_with = "duration_opt_deserialize"
+        )
```

Comment:
By moving `duration_opt_serialize()` and `duration_opt_deserialize()` into a module (say, `duration_opt`), you can use the `with` attribute. Seems a bit neater?